_posts/2023-10-24-referralbyphone.md

```diff
@@ -0,0 +1,98 @@
+
+## CVSS base metrics
+
+* **Attack vector**: network
+* **Attack complexity**: low
+* **Privilege required**: none
+* **User interaction**: none
+* **Scope**: unchanged
+* **Confidentiality**: high
+* **Integrity**: high
+* **Availability**: high
+
+**Vector string**: [CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H](https://nvd.nist.gov/vuln-metrics/cvss/v3-calculator?vector=AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H)
+
+## Possible malicious usage
+
+* Obtain admin access
+* Remove data from the associated PrestaShop
+* Copy/paste data from sensitive tables to FRONT to expose tokens and unlock admins's ajax scripts
+* Rewrite SMTP settings to hijack emails
+
+
+## Proof of concept
+
+```bash
+curl -v -d 'fc=modulemodule=referralbyphone&controller=default&action=CartRuleValidate&ws_voucher=%27;select+0x73656C65637420736C656570283432293B+into+@a;prepare+b+from+@a;execute+b;--' 'https://preprod.X'
+```
+
+
+## Patch from 3.5.1
+
+```diff
+--- 3.5.1/modules/referralbyphone/controllers/front/default.php
++++ XXXXX/modules/referralbyphone/controllers/front/default.php
+        $id_ws_sponsor = Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue(
+-           'SELECT `id_ws_sponsor` FROM `' . _DB_PREFIX_ . 'ws_ref_coupon_rule` WHERE `code` = \'' . $ws_voucher . '\''
++           'SELECT `id_ws_sponsor` FROM `' . _DB_PREFIX_ . 'ws_ref_coupon_rule` WHERE `code` = \'' . pSQL($ws_voucher) . '\''
+        );
+```
+
+## Other recommendations
+
+* It’s recommended to upgrade to the latest version of the module **referralbyphone**.
+* Upgrade PrestaShop to the latest version to disable multiquery executions (separated by “;”) - be warned that this functionality **WILL NOT** protect your SHOP against injection SQL which uses the UNION clause to steal data.
+* Change the default database prefix `ps_` with a new longer, arbitrary prefix. Nevertheless, be warned that this is useless against blackhats with DBA senior skills because of a design vulnerability in DBMS
+* Activate OWASP 942's rules on your WAF (Web application firewall), be warned that you will probably break your backoffice and you will need to pre-configure some bypasses against this set of rules.
+
+## Timeline
+
+| Date | Action |
+|--|--|
+| 2023-07-20 | Issue discovered during a code review by [TouchWeb.fr](https://www.touchweb.fr) |
+| 2023-07-20 | Contact PrestaShop Addons security Team to confirm versions scope by author |
+| 2023-07-20 | PrestaShop Addons security Team confirm versions scope |
+| 2023-07-20 | Relaunch Author for fix |
+| 2023-10-19 | Relaunch Author for fix |
+| 2023-10-19 | PrestaShop Addons security Team confirm author do not patch yet |
```

Comment:
PrestaShop Addons security Team confirms that the author has not yet produced a patch